### PR TITLE
Strip invisible unicode characters from slug

### PR DESCRIPTION
**Pull request by dexterleng**
References: #9826

 Based on the solution [here](https://stackoverflow.com/questions/11598786/how-to-replace-non-printable-unicode-characters-javascript)

Please include a description of your change & check your PR against this list, thanks!

- [x] There's a clear use-case for this code change
- [x] Commit message has a short title & references relevant issues
- [x] The build will pass (run `npm test`)

More info can be found by clicking the "guidelines for contributing" link above.

### CODE REVIEW

**Comment by kevinansfield** [comment]
Hey @dexterleng 👋 Thanks for taking a stab at this! Rather than implementing this only for posts, I think it would be better placed inside the slug generation method on the base model? That way the same problem can't occur for tags and users.

https://github.com/TryGhost/Ghost/blob/master/core/server/models/base/index.js#L838

**Comment by dexterleng** [comment]
Done!

**Comment by ErisDS** [comment]
Can we merge this as a next priority please?

**Comment by gargol** [comment]
@ErisDS can the review wait till next Monday or is it high priority?

**Comment by ErisDS** [comment]
I'd like to get it merged earlier please as I'm blocked by it :)

**Comment by gargol** [comment]
roger, will be reviewed asap :)

File: core/server/models/base/index.js
Comment (gargol) [inline core/server/models/base/index.js]: @dexterleng would be good to have some context on where this string is coming for. Seen similar usages [here](https://github.com/slevithan/xregexp/blob/7a1688742e20b2745436a5a7793104fc17e3e652/tools/output/categories.js#L6). Did you research on any libraries that would already expose such list of non-printable unicode characters? Just thinking of it in terms of future maintenance :thinking:

File: core/test/integration/model/model_tags_spec.js
Comment (gargol) [inline core/test/integration/model/model_tags_spec.js]: This assertion could be returned, in that case, no need to pass `done` around. Something like the following:
```
            return models.Tag.add(tag, context)
                .then(function (newTag) {
                    should.exist(newTag);
                    newTag.get('slug').should.equal('abc');
                });
```

File: core/server/models/base/index.js
Comment (dexterleng) [reply core/server/models/base/index.js]: The stackoverflow [solution here](https://stackoverflow.com/questions/11598786/how-to-replace-non-printable-unicode-characters-javascript). It's from the Xregexp library.

Comment (gargol) [reply core/server/models/base/index.js]: The solution itself is fine :smiley: But what I meant was, we need some way of describing this regexp because it's quite hard to understand at first what exactly it's meant to do. So, adding a source of this regexp and a quick explanation of what it does would help :+1: 

The other concern that I have, is having to maintain this expression with up to date non-printable characters. The current master of Xregexp - https://github.com/slevithan/xregexp/blob/master/tools/output/categories.js#L6 has a different set than the one directly from SO. For now, I think that using the latest set would be enough, but also adding a comment to update it would be good.

File: core/server/models/base/index.js
Comment (gargol) [inline core/server/models/base/index.js]: Was thinking about the location of this method. Probably it could benefit if it was a part of our [security/strings](https://github.com/TryGhost/Ghost/blob/master/core/server/lib/security/string.js) module and if a replacement would happen around [here](https://github.com/TryGhost/Ghost/blob/master/core/server/lib/security/string.js#L12), as it would be more in line with the logic of that module. Would love a second opinion of this @kirrg001

Comment (ErisDS) [reply core/server/models/base/index.js]: It really doesn't matter right now as I am waiting to co-locate the code.

Comment (gargol) [reply core/server/models/base/index.js]: In that case, @dexterleng let's just focus on improvements [here](https://github.com/TryGhost/Ghost/pull/10252#discussion_r241003874) just a bit and will be ready to ship after.

File: core/server/models/base/index.js
Comment (ErisDS) [reply core/server/models/base/index.js]: FWIW we did something very similar previously here: https://github.com/TryGhost/Ghost/commit/9eef2616e4d1ce52f631da89dce47e4d288f47f0#diff-50bf43eef863952813617ae9997538d0R4

File: core/server/models/base/index.js
Comment (dexterleng) [reply core/server/models/base/index.js]: @gargol. Done!

Comment (gargol) [reply core/server/models/base/index.js]: Thanks @dexterleng :+1: Having another :eyes: :smiley:

**Review by naz — APPROVED** [review]
I think it's in good state, given @ErisDS will be relocating the code soon. Thanks for the contribution @dexterleng :tada:

**Pull request merged**
